dll/shellext/cabview/folder.cpp

```diff
@@ -0,0 +1,859 @@
+        }
+    }
+    return E_NOTIMPL;
+}
+
+HRESULT CEnumIDList::Fill(LPCWSTR path, HWND hwnd, SHCONTF contf)
+{
+    FILLCALLBACKDATA data = { this, contf };
+    return ExtractCabinet(path, NULL, EnumFillCallback, &data);
+}
+
+HRESULT CEnumIDList::Fill(PCIDLIST_ABSOLUTE pidl, HWND hwnd, SHCONTF contf)
+{
+    WCHAR path[MAX_PATH];
+    if (SHGetPathFromIDListW(pidl, path))
+        return Fill(path, hwnd, contf);
+    return HRESULT_FROM_WIN32(ERROR_FILE_NOT_FOUND);
+}
+
+IFACEMETHODIMP CCabFolder::GetDefaultColumn(DWORD dwRes, ULONG *pSort, ULONG *pDisplay)
+{
+    if (pSort)
+        *pSort = COL_NAME;
+    if (pDisplay)
+        *pDisplay = COL_NAME;
+    return S_OK;
+}
+
+IFACEMETHODIMP CCabFolder::GetDefaultColumnState(UINT iColumn, SHCOLSTATEF *pcsFlags)
+{
+    if (!pcsFlags || iColumn >= _countof(g_Columns))
+        return E_INVALIDARG;
+    *pcsFlags = g_Columns[iColumn].ColFlags;
+    return S_OK;
+}
+
+IFACEMETHODIMP CCabFolder::GetDisplayNameOf(PCUITEMID_CHILD pidl, DWORD dwFlags, LPSTRRET pName)
+{
+    CABITEM *item = CABITEM::Validate(pidl);
+    if (!item || !pName)
+        return E_INVALIDARG;
+
+    if (dwFlags & SHGDN_FORPARSING)
+    {
+        if (dwFlags & SHGDN_INFOLDER)
+            return StrTo(FLATFOLDER ? item->Path : item->GetName(), *pName);
+
+        WCHAR parent[MAX_PATH];
+        if (!SHGetPathFromIDListW(m_CurDir, parent))
+            return E_FAIL;
+        UINT cch = lstrlenW(parent) + 1 + lstrlenW(item->Path) + 1;
+        pName->uType = STRRET_WSTR;
+        pName->pOleStr = (LPWSTR)SHAlloc(cch * sizeof(WCHAR));
+        if (!pName->pOleStr)
+            return E_OUTOFMEMORY;
+        lstrcpyW(pName->pOleStr, parent);
+        PathAppendW(pName->pOleStr, item->Path);
+        return S_OK;
+    }
+
+    SHFILEINFO fi;
+    DWORD attr = item->IsFolder() ? FILE_ATTRIBUTE_DIRECTORY : 0;
+    UINT flags = SHGFI_DISPLAYNAME | SHGFI_USEFILEATTRIBUTES;
+    if (SHGetFileInfo(item->GetName(), attr, &fi, sizeof(fi), flags))
+        return StrTo(fi.szDisplayName, *pName);
+    return StrTo(item->GetName(), *pName);
+}
+
+HRESULT CCabFolder::GetItemDetails(PCUITEMID_CHILD pidl, UINT iColumn, SHELLDETAILS *psd, VARIANT *pv)
+{
+    HRESULT hr = E_FAIL;
+    STRRET *psr = &psd->str, srvar;
+    CABITEM *item = CABITEM::Validate(pidl);
+    if (!item)
+        return E_INVALIDARG;
+
+    switch (iColumn)
+    {
+        case COL_NAME:
+        {
+            hr = GetDisplayNameOf(pidl, SHGDN_NORMAL | SHGDN_INFOLDER, pv ? &srvar : psr);
+            return SUCCEEDED(hr) && pv ? StrRetToVariantBSTR(&srvar, *pv) : hr;
+        }
+
+        case COL_SIZE:
+        {
+            UINT data = item->Size;
+            if (pv)
+            {
+                V_VT(pv) = VT_UI4;
+                V_UI4(pv) = data;
+            }
+            else
+            {
+                psr->uType = STRRET_CSTR;
+                StrFormatByteSizeA(data, psr->cStr, 260);
```

Comment:
```suggestion
                StrFormatByteSizeA(data, psr->cStr, _countof(psr->cStr));
```

Reply:
Because this struct that was defined in 1995 is likely to change?

Reply:
No, because:
1. 260 looks like magic value;
2. One would have to verify whether it's indeed the size of the static buffer (I had to recheck it again today, somebody else could later);
3. Using countof or sizeof instead of hardcoded values is something we always ask for, and it was already done somewhere below in that file for that very same buffer.